Code/src/main/java/nl/utwente/viskell/ui/components/Lane.java

```diff
@@ -0,0 +1,230 @@
+                    anchor.extendExprGraph(guards, this, surroundingBlocks);
+                });
+            }
+            block.extendExprGraph(guards, this, surroundingBlocks);
+        });
+        
+        return new Pair<>(new Case.Alternative(new ConstructorBinder("()", Collections.EMPTY_LIST), guards), surroundingBlocks);
+    }
+
+    /** Returns the result anchor of this Lane */
+    public ResultAnchor getOutput() {
+        return result;
+    }
+
+    @Override
+    public void refreshAnchorTypes() {
+        // refresh anchor types only once at the start
+        if (!firstPhaseInProgress && !freshAnchorTypes) {
+            freshAnchorTypes = true;
+            
+            TypeScope scope = new TypeScope();
+            arguments.forEach(argument -> argument.refreshType(scope));
+            result.refreshAnchorType(scope);
+        }
+    }
+    
+    @Override
+    public final void handleConnectionChanges(boolean finalPhase) {
+        // avoid doing extra work and infinite recursion
+        if ((!finalPhase && !firstPhaseInProgress) || (finalPhase && !finalPhaseInProgress)) {
+            if (!finalPhase) {
+                // in first phase ensure that anchor types are refreshed if propagating from the outside
+                refreshAnchorTypes();
+
+                firstPhaseInProgress = true;
+                finalPhaseInProgress = false;
+            }
+            else {
+                firstPhaseInProgress = false;
+                finalPhaseInProgress = true;
+            }
+
+            freshAnchorTypes = false;
+            
+            // first propagate up from the result anchor
+            result.getConnection().ifPresent(c -> c.handleConnectionChangesUpwards(finalPhase));
+            
+            // also propagate in from above in case the lane is partially connected 
+            arguments.forEach(argument -> {
+                argument.getOppositeAnchors().forEach(anchor -> {
+                    anchor.handleConnectionChanges(finalPhase);
+                    // take the type of argument connections into account even if the connected block is being processed
+                    anchor.getConnection().ifPresent(connection -> connection.handleConnectionChangesUpwards(finalPhase));
+                });
+            });
+    
+            // propagate internal type changes outwards
+            parent.handleConnectionChanges(finalPhase);
+        }
+    }
+
+    /** Called when the VisualState changed. */
+    public void invalidateVisualState() {
+        // TODO update anchors when they get a type label       
+    }
+    
+    /** Add and initializes a resizer element to this block */
+    private void setupResizer() {
+        resizer = new Polygon();
+        resizer.getPoints().addAll(new Double[]{20.0, 20.0, 20.0, 0.0, 0.0, 20.0});
+        resizer.setFill(Color.BLUE);
+
+        resizer.setManaged(false);
+        this.getChildren().add(resizer);
+        resizer.relocate(300-20, 400-20);
```

Comment:
This is copied from DefinitionBlock - might be worth extracting or moving to a common superclass.